### PR TITLE
API-5121 - Added Fraud Prevention link to Left hand navigation

### PR DESCRIPTION
**Pull request by anjumabbas5**

### CODE REVIEW

File: app/config/ApplicationConfig.scala [hunk 1]
```diff
@@ -94,6 +94,11 @@ class ApplicationConfig @Inject()(config: Configuration, runMode: RunMode) exten
   val ppnsSandboxApiKey = getConfString("push-pull-notifications-api-sandbox.api-key", "")
   val ppnsSandboxAuthorizationKey = getConfString("push-pull-notifications-api-sandbox.authorizationKey", "")
 
+  // Fraud Prevention Headers
+  lazy val fraudPreventionLinkVisible = getConfigDefaulted("fraudPreventionLinkVisible", false)
```
Comment (AndySpaven) [inline app/config/ApplicationConfig.scala:98]: Can we loose the defaulted.  We're trying to move to HOCON so these go in application.conf instead and blow up if you loose them.  I know the other config stull has defaults.  But they are also yuk.

File: app/config/ApplicationConfig.scala [hunk 2]
```diff
@@ -94,6 +94,11 @@ class ApplicationConfig @Inject()(config: Configuration, runMode: RunMode) exten
...
+  lazy val fraudPreventionApis = getConfigDefaulted[Seq[String]]("fraudPreventionApis", List.empty)
```
Comment (AndySpaven) [inline app/config/ApplicationConfig.scala:99]: And can we put this in a separate config item and add the provider for it.  We're also trying to break down large config.

File: conf/application.conf [hunk 3]
```diff
@@ -124,6 +124,11 @@ applicationCheck {
   canUseNewUpliftJourney = "OnDemand"
 }
 
+fraudPreventionLinkVisible = false
```
Comment (AndySpaven) [inline conf/application.conf:127]: can we structure this as fraudPrevention {...}

**Comment by peteslater-ee** [comment]
Instead of having to add `hasFraudPreventionHeaders = false` to every instance of `ApplicationViewModel` in the tests, it may have been better to pull out a method for building those models, re-factor the code to use it and add the `hasFraudPreventionHeaders = false` in there. Once.

File: conf/application.conf (hunk 3 above)
Comment (AndySpaven) [reply conf/application.conf:127]: or better fraudPreventionHeaders or something

File: test/views/DetailsSpec.scala [hunk 4]
```diff
@@ -66,13 +66,13 @@ class DetailsSpec
         "Show Production when environment is Production" in {
           when(appConfig.nameOfPrincipalEnvironment).thenReturn("Production")
           when(appConfig.nameOfSubordinateEnvironment).thenReturn("Sandbox")
-          val page = Page(detailsView(ApplicationViewModel(application, hasSubscriptionsFields = false, hasPpnsFields = false)))
+          val page = Page(detailsView(ApplicationViewModel(application, hasSubscriptionsFields = false, hasPpnsFields = false, hasFraudPreventionHeaders = false)))
```
Comment (AndySpaven) [inline test/views/DetailsSpec.scala:69]: This is getting a bit long.   Time for a local createPage method?

File: app/controllers/Controller.scala [hunk 5]
```diff
@@ -54,36 +53,36 @@ case class UserRequest[A](developerSession: DeveloperSession, request: MessagesR
...
-    request: MessagesRequest[A]
-) extends MessagesRequest[A](request, request.messagesApi) {
+                                  application: Application,
+                                  deployedTo: Environment,
+                                  subscriptions: List[APISubscriptionStatus],
+                                  openAccessApis: Map[ApiContext, ApiData],
+                                  role: CollaboratorRole,
+                                  user: DeveloperSession,
+                                  request: MessagesRequest[A]
+                                ) extends MessagesRequest[A](request, request.messagesApi) {
```
Comment (peteslater-ee) [inline app/controllers/Controller.scala:63]: My eyes... the spacing.

File: conf/application.conf [hunk 6]
```diff
@@ -124,10 +124,12 @@ applicationCheck {
   canUseNewUpliftJourney = "OnDemand"
 }
 
-fraudPreventionLinkVisible = false
-fraudPreventionApis = ["vat-api", "self-assessment-api"]
-fraudPreventionUrl = "https://developer.service.hmrc.gov.uk/api-documentation/docs/reference-guide#fraud-prevention"
 
+fraudPreventionLink {
+  linkEnabled = true
```
Comment (peteslater-ee) [inline conf/application.conf:129]: enabled

File: conf/application.conf [hunk 7]
```diff
@@ -124,10 +124,12 @@ applicationCheck {
...
+  apisWithFraudPrevention = ["vat-api", "self-assessment-api"]
+  linkUrl = "https://developer.service.hmrc.gov.uk/api-documentation/docs/reference-guide#fraud-prevention"
```
Comment (peteslater-ee) [inline conf/application.conf:131]: uri/url

Comment (mattclark-zerogravit) [reply conf/application.conf]: done

File: conf/application.conf (hunk 6 above)
Comment (mattclark-zerogravit) [reply conf/application.conf]: done

File: app/controllers/Controller.scala [hunk 8]
```diff
@@ -98,23 +98,23 @@ abstract class LoggedInController(mcc: MessagesControllerComponents) extends Bas
 
 abstract class ApplicationController(mcc: MessagesControllerComponents) extends LoggedInController(mcc) with ActionBuilders {
   val applicationService: ApplicationService
-  val fraudPreventionConfigProvider: FraudPreventionConfigProvider
+  val fraudPreventionConfig: config.FraudPreventionConfig
```
Comment (peteslater-ee) [inline app/controllers/Controller.scala:101]: No need for the `config` prefix here if you import it properly at the top of the file.

**Review by AndySpaven — APPROVED** [review]
Let's go with this.

**Pull request merged**